src/transaction/boot_sr.c

```diff
@@ -3825,9 +3827,12 @@ xboot_check_db_consistency (THREAD_ENTRY * thread_p, int check_flag, OID * oids,
   bool repair = check_flag & CHECKDB_REPAIR;
   int error_code = NO_ERROR;
 
-  disk_lock_extend ();
-  error_code = boot_check_permanent_volumes (thread_p);
-  disk_unlock_extend ();
+  if (!is_tran_server_with_remote_storage ())
```

Comment:
Should this function be called in TSRS context?

Reply:
Yes, it should be called in TSRS context. TSRS cannot do `boot_check_permanent_volumes ()` because this function verifies information in database is matching the information in file cache; but TSRS does not keep any files in the file cache.

The other checks verify information in the database is consistent (e.g. data in heap files is consistent with data in indexes). These checks can be done only on TS, because PS is clueless and cannot interpret the information in its data pages.

Reply:
So the question was clueless. My bad.